### PR TITLE
Fix the React proxy settings to point to our web-ui backend

### PR DESCRIPTION
**Pull request by inickles-grapl**
### Which issue does this PR correspond to?

Not sure if filed, but the development loop for frontend (engagement-view) was essentially broken per standard processes for React development. TIL.

### What changes does this PR make to Grapl? Why?

This configures the React middleware proxy to point to the backend web server in our local deployments (grapl-web-ui) for our auth and graphql apis, allowing frontend devs to run `yarn start` for development.

1. Fix the React settings to point to our backend web server.
2. Add a Make target to the engagement-view Makefile for running `yarn start` in a dev container.

### How were these changes tested?

1. `$ KEEP_TEST_ENV=1 make test-e2e`
2. `$ cd src/js/engagement_view`
3. `$ make start`
4. Browse to http://localhost:3000
5. login with test user credentials
6. See the engagement from the E2E tests
7. Make a simple color change to one of the component things (I forget which), save the file.
8. Observed the color updated in the browser without any other input.

### CODE REVIEW

**Review by twunderlich-grapl — APPROVED** [review]

File: src/js/engagement_view/src/setupProxy.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+const { createProxyMiddleware } = require('http-proxy-middleware');
```
Comment (christophermaier) [inline src/js/engagement_view/src/setupProxy.js:1]: Is the change from `*.tsx` to `*.js` in the filename significant? This seems (to my admittedly shallow knowledge of the dark inner workings of Javascript and Typescript) to be the same file but with a single additional proxy middleware.

File: src/js/engagement_view/tsconfig.json [hunk 2]
```diff
@@ -25,5 +25,5 @@
   },
   "include": [
     "src/**/*.ts"
-, "src/tests/graphDisplay.test.tsx", "src/setupProxy.tsx"  ]
+, "src/tests/graphDisplay.test.tsx" ]
```
Comment (christophermaier) [inline src/js/engagement_view/tsconfig.json:28]: Does the new `setupProxy.js` file get magically picked up somehow? Is the name special?

File: src/js/engagement_view/src/setupProxy.js (hunk 1 above)
Comment (inickles-grapl) [reply src/js/engagement_view/src/setupProxy.js:1]: Unfortunately, it is. It must be. I spent **far** too much time thinking it should work as a Typescript file, but no dice.

File: src/js/engagement_view/tsconfig.json (hunk 2 above)
Comment (inickles-grapl) [reply src/js/engagement_view/tsconfig.json:28]: Yeah, unfortunately. And so it's worth me mentioning this somewhere.

I think I'll document a) this behavior, and b) how to use it with Grapl in the README for this project. I'll add that to this PR.

File: src/js/engagement_view/src/setupProxy.js (hunk 1 above)
Comment (christophermaier) [reply src/js/engagement_view/src/setupProxy.js:1]: That's probably worth an explanatory comment to that effect, as well as any link to relevant background information or documentation.

File: src/js/engagement_view/tsconfig.json (hunk 2 above)
Comment (christophermaier) [reply src/js/engagement_view/tsconfig.json:28]: 🤢 Ugh magic

Thanks for documenting these horrors for the rest of us :bow:

File: src/js/engagement_view/Makefile [hunk 3]
```diff
@@ -76,6 +76,12 @@ shell: build-image
 shell: ## Drop into a shell inside the build container
 	$(DOCKER_RUN) bash
 
+.PHONY: start
+start: extra_docker_run_args := --interactive --tty --network host
```
[the remark this answers is not in the document]
Comment (inickles-grapl) [reply src/js/engagement_view/Makefile:80]: Thank you for that *tips fedora*

File: src/js/engagement_view/tsconfig.json (hunk 2 above)
Comment (inickles-grapl) [reply src/js/engagement_view/tsconfig.json:28]: Okay, I decided against updating the README for now, but there's a bunch of info in src/setupProxy.js.

Comment (christophermaier) [reply src/js/engagement_view/tsconfig.json:28]: Nice addition; thanks for that 👍

**Review by christophermaier — APPROVED** [review]

**Pull request merged**